enterprise/backend/test/metabase_enterprise/sandbox/api/card_test.clj

```diff
@@ -102,6 +102,35 @@
                                                                                      (api.card-test/mbql-count-query db table))
                                              :collection_id (u/the-id collection)))))))))))
 
+(deftest users-with-data-access-and-query-create-may-access-cards
+  (mt/with-temp [:model/User                       {user-id :id} {}
+                 :model/PermissionsGroup           group         {}
+                 :model/PermissionsGroupMembership _             {:user_id  user-id
+                                                                  :group_id (u/the-id group)}
+                 :model/Card                       card          {:name "Some Name" :dataset_query {:database (mt/id),
+                                                                                                    :type :query,
+                                                                                                    :query {:source-table (mt/id :venues)}}}]
+    (let [cases [[:unrestricted           :query-builder-and-native "Request Permitted"]
+                 [:unrestricted           :query-builder            "Request Permitted"]
+                 [:unrestricted           :no                       "Request Permitted"]
+                 [:legacy-no-self-service :query-builder-and-native "Request Permitted"]
+                 [:legacy-no-self-service :query-builder            "Request Permitted"]
+                 [:legacy-no-self-service :no                       "You do not have permissions to run this query."]
+                 [:blocked                :query-builder-and-native "Request Permitted"]
+                 [:blocked                :query-builder            "Request Permitted"]
```

Comment:
I think these are actually an impossible combo on the table level. db level too, but we are testing tables here.